addons/website/tools.py

```diff
@@ -167,3 +168,20 @@ def text_from_html(html_fragment, collapse_whitespace=False):
     if collapse_whitespace:
         content = re.sub('\\s+', ' ', content).strip()
     return content
+
+def get_base_domain(url):
+    """
+    Returns the domain of a given url without the scheme and the www. and the
+    final '/' if any.
+
+    :param url: url from which the domain must be extracted
+
+    :return: domain of the url
+    """
+    if not url:
+        return ''
+
+    url = werkzeug.urls.url_parse(url).netloc
+    if url.startswith('www.'):
+        url = url[4:]
+    return url
```

Comment:
What I would do, depending of how you judge the usefulness of this method is:
- _If judged really useful but also judged worth an util rather than copying it where needed_: Ok to keep it in website, it will saves us an approval from another team and it's not the most useful util we made so they can live without it.
Still, I'd make the `ẁww.` strip conditional like a method param `strip_www=False` and I would use the util here too: https://github.com/odoo/odoo/blame/f9d5d1919b5794623a2c6296135f0692f8979e08/addons/website/models/website.py#L973
https://github.com/odoo/odoo/blame/f9d5d1919b5794623a2c6296135f0692f8979e08/addons/website/controllers/main.py#L141
- _If judged not really useful_: I would make it a sub method of `should_index_url` and call it in there directly.

I doubt it's really usual to want to have the base domain with `www` strip, but I'll let you decide about that :+1: pick what you prefer (or suggest something else)